examples/correlated_ou/analysis.jl

```diff
@@ -9,91 +9,90 @@ using Statistics
 using TOML
 
 using CliMAgen
+
 package_dir = pkgdir(CliMAgen)
-include(joinpath(package_dir,"examples/utils_data.jl"))
-include(joinpath(package_dir,"examples/utils_analysis.jl"))
+include(joinpath(package_dir, "examples/utils_data.jl"))
+include(joinpath(package_dir, "examples/utils_analysis.jl"))
 
 function run_analysis(params; FT=Float32, logger=nothing)
     # unpack params
     savedir = params.experiment.savedir
     rngseed = params.experiment.rngseed
     nogpu = params.experiment.nogpu
-    
+
     resolution = params.data.resolution
     fraction = params.data.fraction
     standard_scaling = params.data.standard_scaling
     preprocess_params_file = joinpath(savedir, "preprocessing_standard_scaling_$standard_scaling.jld2")
-    
-    inchannels = params.model.noised_channels
+
+    inchannels = params.model.inchannels
+
     nsamples = params.sampling.nsamples
-    nimages = params.sampling.nimages
     nsteps = params.sampling.nsteps
     sampler = params.sampling.sampler
-    tilesize_sampling = params.sampling.tilesize
 
     # set up rng
     rngseed > 0 && Random.seed!(rngseed)
 
     # set up device
     if !nogpu && CUDA.has_cuda()
-        device = Flux.gpu
+        dev = Flux.gpu
         @info "Sampling on GPU"
     else
-        device = Flux.cpu
+        dev = Flux.cpu
         @info "Sampling on CPU"
     end
 
     # set up dataset
-    dl, _ = get_data_correlated_ou2d(
+    dl, _ = get_data_correlated_ou2d_timeseries(
         nsamples;
         f = fraction,
         resolution=resolution,
+        ntime=16,
         FT=Float32,
         standard_scaling = standard_scaling,
         read = true,
         save = false,
         preprocess_params_file,
         rng=Random.GLOBAL_RNG
     )
-    xtrain = first(dl)
+    xtrain, dl = Iterators.peel(dl)
 
     # set up model
     checkpoint_path = joinpath(savedir, "checkpoint.bson")
     BSON.@load checkpoint_path model model_smooth opt opt_smooth
-    model = device(model)
-    
+    model = dev(model)
+
     # sample from the trained model
-    time_steps, Δt, init_x = setup_sampler(
-        model,
-        device,
-        tilesize_sampling,
-        inchannels;
-        num_images=nsamples,
-        num_steps=nsteps,
-    )
-    if sampler == "euler"
-        samples = Euler_Maruyama_sampler(model, init_x, time_steps, Δt)
-    elseif sampler == "pc"
-        samples = predictor_corrector_sampler(model, init_x, time_steps, Δt)
-    end
+    t = ones(FT, nsamples) |> dev
+    init_z = randn(FT, (resolution, resolution, 16, 1, nsamples)) |> dev
+    _, σ_T = CliMAgen.marginal_prob(model, zero(init_z), t)
+    init_x = (σ_T .* init_z)
+    time_steps = LinRange(1.0f0, 1.0f-3, nsteps)
+    Δt = time_steps[1] - time_steps[2]
+    samples = Euler_Maruyama_sampler(model, init_x, time_steps, Δt, nspatial=3)
```

Comment:
it is it worth making setup_sampler support nspatial as an argument? or pass nx, ny, nt?

Reply:
We can consider that in our overhaul maybe. This PR is getting a bit big :).